okhttp-apache/src/main/java/com/squareup/okhttp/apache/OkApacheClient.java

```diff
@@ -39,12 +39,22 @@
  * and others.
  */
 public final class OkApacheClient implements HttpClient {
-  private static Request transformRequest(HttpRequest request) {
+  private static Request transformRequest(HttpHost httpHost, HttpRequest request) {
     Request.Builder builder = new Request.Builder();
 
     RequestLine requestLine = request.getRequestLine();
     String method = requestLine.getMethod();
-    builder.url(requestLine.getUri());
+    String url = requestLine.getUri();
+    if (httpHost != null) {
+      try {
+        URI uri = URI.create(url);
```

Comment:
Wanna use our new `HttpUrl` hotness for construction here? That'll support more URLs.